src/main/java/org/opensrp/web/config/security/filter/XssPreventionRequestWrapper.java

```diff
@@ -112,11 +113,27 @@ private static boolean isValidJSON(final String json) throws IOException {
 	private class ResettableServletInputStream extends ServletInputStream {
 
 		private InputStream stream;
+		private ReadListener readListener;
 
 		@Override
 		public int read() throws IOException {
 			return stream.read();
 		}
+
+		@Override
+		public boolean isFinished() {
+			return false;
+		}
+
+		@Override
+		public boolean isReady() {
+			return false;
+		}
+
+		@Override
+		public void setReadListener(ReadListener readListener) {
+			this.readListener = readListener;
+		}
```

Comment:
Add the same for these, alternatively, see if you can some solid implementation online that we can trust and use